src/lib/swapper/swappers/ArbitrumBridgeSwapper/utils/fetchArbitrumBridgeSwap.ts

```diff
@@ -0,0 +1,63 @@
+import { Erc20Bridger, EthBridger, getL2Network } from '@arbitrum/sdk'
+import type {
+  L1ToL2TransactionRequest,
+  L2ToL1TransactionRequest,
+} from '@arbitrum/sdk/dist/lib/dataEntities/transactionRequest'
+import { ethAssetId, ethChainId, fromAssetId } from '@shapeshiftoss/caip'
+import { isEvmChainId } from '@shapeshiftoss/chain-adapters'
+import { type Asset, KnownChainIds } from '@shapeshiftoss/types'
+import { BigNumber } from 'ethers5'
+import { arbitrum } from 'viem/chains'
+import { getEthersV5Provider } from 'lib/ethersProviderSingleton'
+
+export type FetchArbitrumBridgeSwapInput = {
+  affiliateBps: string
+  buyAsset: Asset
+  receiveAddress: string
+  sellAmountIncludingProtocolFeesCryptoBaseUnit: string
+  sellAsset: Asset
+  maximumSlippageDecimalPercentage: string
+  sendAddress: string
+}
+
+export const fetchArbitrumBridgeSwap = async ({
+  buyAsset,
+  sellAmountIncludingProtocolFeesCryptoBaseUnit,
+  sellAsset,
+  sendAddress,
+  receiveAddress,
+}: FetchArbitrumBridgeSwapInput): Promise<
+  Omit<L1ToL2TransactionRequest | L2ToL1TransactionRequest, 'retryableData'>
+> => {
+  const l2Network = await getL2Network(arbitrum.id)
+  const isDeposit = sellAsset.chainId === ethChainId
+  const isEthBridge = isDeposit ? sellAsset.assetId === ethAssetId : buyAsset.assetId === ethAssetId
+
+  const bridger = isEthBridge ? new EthBridger(l2Network) : new Erc20Bridger(l2Network)
+
+  if (!(isEvmChainId(sellAsset.chainId) && isEvmChainId(buyAsset.chainId))) {
```

Comment:
`assertValidTrade` is available for reuse

Reply:
c9b315ed97